modules/game/src/main/UciMemo.scala

```diff
@@ -30,33 +30,38 @@ final class UciMemo(gameRepo: GameRepo)(implicit ec: scala.concurrent.ExecutionC
       action.player
     )
 
-  def set(game: Game, actionStrs: ActionStrs) =
-    cache.put(game.id, actionStrs)
+  private def set(game: Game, uciStrs: ActionStrs) =
+    cache.put(game.id, uciStrs)
+
+  def set(game: Game, fen: Option[FEN]) =
+    for {
+      uciStrs <- compute(game, maxTurns, fen).toFuture
+    } yield cache.put(game.id, uciStrs)
 
   def set(game: Game) =
-    cache.put(game.id, game.actionStrs)
+    for {
+      uciStrs <- compute(game, maxTurns)
+    } yield cache.put(game.id, uciStrs)
 
   def get(game: Game, max: Int = maxTurns): Fu[ActionStrs] =
-    cache getIfPresent game.id filter { actionStrs =>
-      actionStrs.size.min(max) == game.actionStrs.size.min(max)
+    cache.getIfPresent(game.id).filter { uciStrs =>
+      uciStrs.size.min(max) == game.actionStrs.size.min(max)
     } match {
-      case Some(actionStrs) => fuccess(actionStrs)
-      case _                => compute(game, max) addEffect { set(game, _) }
+      case Some(uciStrs) => fuccess(uciStrs)
+      case _             => compute(game, max).addEffect { set(game, _) }
     }
 
-  //def drop(game: Game, nb: Int) = {
-  //  val current = ~cache.getIfPresent(game.id)
-  //  cache.put(game.id, current.take(current.size - nb))
-  //}
+  private def compute(game: Game, max: Int, fen: Option[FEN]): cats.data.Validated[String, ActionStrs] =
```

Comment:
I feel like this should be called `gameToUci` or something. As that's what it is. Having it the same name as the other `compute` function is a bit strange and confusing imo